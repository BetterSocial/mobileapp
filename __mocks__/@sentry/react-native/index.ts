export const addBreadcrumb = jest.fn();
