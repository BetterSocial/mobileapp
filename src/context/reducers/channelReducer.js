import {SET_CHANNEL} from '../Types';

const channelState = {
  channel: null,
};
const channelReducer = (state = channelState, action) => {
  switch (action.type) {
    case SET_CHANNEL:
      return {
        ...state,
        channel: action.payload,
      };
    default:
      return state;
  }
};
export {channelReducer, channelState};
