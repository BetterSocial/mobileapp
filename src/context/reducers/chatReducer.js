import {SET_REPLY_TARGET} from '../Types';

const chatState = {
  replyTarget: null
};
const chatReducer = (state = chatState, action) => {
  switch (action.type) {
    case SET_REPLY_TARGET:
      return {
        ...state,
        replyTarget: action.payload
      };
    default:
      return state;
  }
};
export {chatReducer, chatState};
