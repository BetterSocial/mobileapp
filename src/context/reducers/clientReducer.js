import {SET_CHAT_CLIENT, SET_CLIENT} from '../Types';

const clientState = {
  client: null,
  chatClient: null
};
const clientReducer = (state = clientState, action) => {
  switch (action.type) {
    case SET_CLIENT:
      return {
        ...state,
        client: action.payload,
      };
    case SET_CHAT_CLIENT:
      return {
        ...state,
        chatClient: action.payload
      }
    default:
      return state;
  }
};
export {clientReducer, clientState};
