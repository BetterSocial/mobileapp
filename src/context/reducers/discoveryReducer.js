/* eslint-disable no-case-declarations */
import {
  DISCOVERY_RESET,
  DISCOVERY_SET_DATA,
  DISCOVERY_SET_DATA_DOMAINS,
  DISCOVERY_SET_DATA_NEWS,
  DISCOVERY_SET_DATA_TOPICS,
  DISCOVERY_SET_DATA_USERS,
  DISCOVERY_SET_FIRST_TIME_OPEN,
  DISCOVERY_SET_FOCUS,
  DISCOVERY_SET_INITIAL_TOPICS,
  DISCOVERY_SET_INITIAL_USERS,
  DISCOVERY_SET_LOADING_DATA,
  DISCOVERY_SET_LOADING_DATA_DOMAIN,
  DISCOVERY_SET_LOADING_DATA_NEWS,
  DISCOVERY_SET_LOADING_DATA_TOPIC,
  DISCOVERY_SET_LOADING_DATA_USER,
  DISCOVERY_SET_NEW_FOLLOWED_DOMAIN,
  DISCOVERY_SET_NEW_FOLLOWED_USER,
  DISCOVERY_SET_NEW_UNFOLLOWED_DOMAIN,
  DISCOVERY_SET_NEW_UNFOLLOWED_USER,
  DISCOVERY_SET_RECENT_SEARCH,
} from '../Types';

/**
 * @typedef {Object} DiscoveryState
 * @property {Object[]} initialUsers
 * @property {Object[]} initialTopics
 * @property {Object[]} followedUsers
 * @property {Object[]} unfollowedUsers
 * @property {Object[]} followedTopic
 * @property {Object[]} unfollowedTopic
 * @property {Object[]} followedDomains
 * @property {Object[]} unfollowedDomains
 * @property {Object[]} news
 * @property {Boolean} isLoadingDiscovery
 */
const discoveryState = {
  initialUsers: [],
  initialTopics: [],
  followedUsers: [],
  unfollowedUsers: [],
  followedTopic: [],
  unfollowedTopic: [],
  followedDomains: [],
  unfollowedDomains: [],
  news: [],
  recentSearch: [],
  isLoadingDiscoveryUser: false,
  isLoadingDiscoveryDomain: false,
  isLoadingDiscoveryTopic: false,
  isLoadingDiscoveryNews: false,
  isFirstTimeOpen: true,
  isFocus: true,
};

/**
 *
 * @typedef {Object} DiscoveryReducerActionPayload
 * @property {import('../../service/discovery').FetchDiscoveryDataResponse} discovery
 */
/**
 *
 * @typedef {Object} DiscoveryReducerAction
 * @property {DiscoveryReducerActionPayload} payload
 */
/**
 *
 * @param {Object} state
 * @param {DiscoveryReducerAction} action
 * @returns
 */
const discoveryReducer = (state = discoveryState, action) => {
  switch (action.type) {
    // case DISCOVERY_SET_DATA:
    //   const {
    //     followedUsers, unfollowedUsers, followedDomains,
    //     unfollowedDomains, followedTopic, unfollowedTopic, news,
    //   } = action.payload.discovery;
    //   return {
    //     ...state,
    //     followedUsers,
    //     unfollowedUsers,
    //     followedDomains,
    //     unfollowedDomains,
    //     followedTopic,
    //     unfollowedTopic,
    //     news,
    //   };

    case DISCOVERY_RESET:
      return {
        ...state,
        isFirstTimeOpen: true,
        isLoadingDiscoveryUser: false,
        isLoadingDiscoveryDomain: false,
        isLoadingDiscoveryTopic: false,
        isLoadingDiscoveryNews: false,
      };

    case DISCOVERY_SET_INITIAL_TOPICS:
      return {
        ...state,
        initialTopics: action.payload,
      };

    case DISCOVERY_SET_INITIAL_USERS:
      return {
        ...state,
        initialUsers: action.payload,
      };

    case DISCOVERY_SET_DATA_USERS:
      const {
        followedUsers, unfollowedUsers,
      } = action.payload.discovery;
      return {
        ...state,
        followedUsers,
        unfollowedUsers,
      };

    case DISCOVERY_SET_DATA_DOMAINS:
      const {
        followedDomains, unfollowedDomains,
      } = action.payload.discovery;
      return {
        ...state,
        followedDomains,
        unfollowedDomains,
      };

    case DISCOVERY_SET_DATA_TOPICS:
      const {
        followedTopic, unfollowedTopic,
      } = action.payload.discovery;
      return {
        ...state,
        followedTopic,
        unfollowedTopic,
      };

    case DISCOVERY_SET_DATA_NEWS:
      const {
        news,
      } = action.payload.discovery;
      return {
        ...state,
        news,
      };

    case DISCOVERY_SET_FOCUS:
      return {
        ...state,
        isFocus: action.payload,
      };

    case DISCOVERY_SET_FIRST_TIME_OPEN:
      return {
        ...state,
        isFirstTimeOpen: action.payload,
      };

    case DISCOVERY_SET_RECENT_SEARCH:
      return {
        ...state,
        recentSearch: action.payload,
      };

    case DISCOVERY_SET_LOADING_DATA:
      return {
        ...state,
        isLoadingDiscoveryUser: action.payload,
        isLoadingDiscoveryDomain: action.payload,
        isLoadingDiscoveryTopic: action.payload,
        isLoadingDiscoveryNews: action.payload,
      };

    case DISCOVERY_SET_LOADING_DATA_USER:
      return {
        ...state,
        isLoadingDiscoveryUser: action.payload,
      };

    case DISCOVERY_SET_LOADING_DATA_TOPIC:
      return {
        ...state,
        isLoadingDiscoveryTopic: action.payload,
      };

    case DISCOVERY_SET_LOADING_DATA_DOMAIN:
      return {
        ...state,
        isLoadingDiscoveryDomain: action.payload,
      };

    case DISCOVERY_SET_LOADING_DATA_NEWS:
      return {
        ...state,
        isLoadingDiscoveryNews: action.payload,
      };

    case DISCOVERY_SET_NEW_FOLLOWED_USER:
      return {
        ...state,
        followedUsers: action.payload.newFollowedUsers,
      };

    case DISCOVERY_SET_NEW_UNFOLLOWED_USER:
      return {
        ...state,
        unfollowedUsers: action.payload.newUnfollowedUsers,
      };

    case DISCOVERY_SET_NEW_FOLLOWED_DOMAIN:
      return {
        ...state,
        followedDomains: action.payload.newFollowedDomains,
      };

    case DISCOVERY_SET_NEW_UNFOLLOWED_DOMAIN:
      return {
        ...state,
        unfollowedDomains: action.payload.newUnfollowedDomains,
      };
    default:
      return state;
  }
};
export { discoveryReducer, discoveryState };
