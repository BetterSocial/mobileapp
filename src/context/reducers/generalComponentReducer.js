import {
  DISCOVERY_SET_SEARCH_BAR,
} from '../Types';

const generalComponentState = {
  discoverySearchBarText: '',
};
const generalComponentReducer = (state = generalComponentState, action) => {
  switch (action.type) {
    case DISCOVERY_SET_SEARCH_BAR:
      return {
        ...state,
        discoverySearchBarText: action.payload,
      };
    default:
      return state;
  }
};
export { generalComponentReducer, generalComponentState };
