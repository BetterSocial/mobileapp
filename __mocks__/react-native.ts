import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN