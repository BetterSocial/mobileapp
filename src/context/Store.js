import * as React from 'react';

import {channelReducer, channelState} from './reducers/channelReducer';
import {clientReducer, clientState} from './reducers/clientReducer';
import {discoveryReducer, discoveryState} from './reducers/discoveryReducer';
import {domainReducer, domainState} from './reducers/domainReducer';
import {feedsReducer, feedsState} from './reducers/FeedReducer';
import {followingReducer, followingState} from './reducers/followingReducer';
import {generalComponentReducer, generalComponentState} from './reducers/generalComponentReducer';
import {groupChatReducer, groupChatState} from './reducers/groupChat';
import {initialStartupReducer, initialStartupState} from './reducers/intialStartupReducer';
import {localCommunityReducer, localCommunityState} from './reducers/localComunityReducer';
import {myProfileFeedReducer, myProfileFeedState} from './reducers/myProfileFeedReducer';
import {myProfileReducer, myProfileState} from './reducers/myProfileReducer';
import {newsReducer, newsState} from './reducers/newsReducer';
import {otherProfileFeedReducer, otherProfileFeedState} from './reducers/OtherProfileFeedReducer';
import {topicsReducer, topicsState} from './reducers/topicsReducer';
import {unReadMessageReducer, unReadMessageState} from './reducers/unReadMessageReducer';
import {usersReducer, usersState} from './reducers/userReducer';
import {commentReducer, commentState} from './reducers/commentReducer';

/**
 *
 * @typedef {Object} MainContext
 * @property {import('./reducers/discoveryReducer').DiscoveryState} discovery
 */
const Store = ({children}) => {
  const rootReducer = {
    channel: React.useReducer(channelReducer, channelState),
    client: React.useReducer(clientReducer, clientState),
    discovery: React.useReducer(discoveryReducer, discoveryState),
    feeds: React.useReducer(feedsReducer, feedsState),
    following: React.useReducer(followingReducer, followingState),
    generalComponent: React.useReducer(generalComponentReducer, generalComponentState),
    groupChat: React.useReducer(groupChatReducer, groupChatState),
    localCommunity: React.useReducer(localCommunityReducer, localCommunityState),
    myProfileFeed: React.useReducer(myProfileFeedReducer, myProfileFeedState),
    otherProfileFeed: React.useReducer(otherProfileFeedReducer, otherProfileFeedState),
    news: React.useReducer(newsReducer, newsState),
    profile: React.useReducer(myProfileReducer, myProfileState),
    topics: React.useReducer(topicsReducer, topicsState),
    unReadMessage: React.useReducer(unReadMessageReducer, unReadMessageState),
    users: React.useReducer(usersReducer, usersState),
    domains: React.useReducer(domainReducer, domainState),
    initialStartup: React.useReducer(initialStartupReducer, initialStartupState),
    comments: React.useReducer(commentReducer, commentState)
  };
  return <Context.Provider value={rootReducer}>{children}</Context.Provider>;
};
export const Context = React.createContext(null);
export const useAppContext = () => React.useContext(Context);

export default Store;
