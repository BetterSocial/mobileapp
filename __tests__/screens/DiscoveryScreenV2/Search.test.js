import * as React from 'react'
import renderer from 'react-test-renderer'
import { Pressable, View } from 'react-native'
import { cleanup, fireEvent, render } from '@testing-library/react-native'

import DiscoverySearch from '../../../src/screens/DiscoveryScreenV2/elements/Search'
import TestIdConstant from '../../../src/utils/testId'
import { Context } from '../../../src/context/Store'
import { discoveryState } from '../../../src/context/reducers/discoveryReducer'
import { generalComponentState } from '../../../src/context/reducers/generalComponentReducer'

beforeEach(cleanup)

jest.useFakeTimers('modern')

describe('Testing Discovery Screen V2', () => {
    describe('Testing Discovery Search Bar', () => {
        it('Match Snapshot', () => {
            const contextValue = {
                generalComponent: [generalComponentState, () => jest.fn()],
                discovery: [discoveryState, () => jest.fn()]
            }

            const tree = renderer.create(
                <Context.Provider value={contextValue}>
                    <DiscoverySearch />
                </Context.Provider>
            ).toJSON()
            expect(tree).toMatchSnapshot()
        })

        it('Change text when given', () => {
            const discoveryDispatch = jest.fn()
            const generalDispatch = jest.fn()
            const setSearchText = jest.fn()

            const contextValue = {
                generalComponent: [generalComponentState, generalDispatch],
                discovery: [discoveryState, discoveryDispatch]
            }

            const { getByTestId, getByText } = render(
                <Context.Provider value={contextValue}>
                    <DiscoverySearch setSearchText={setSearchText}/>
                </Context.Provider>
            )

            fireEvent.changeText(getByTestId(TestIdConstant.discoveryScreenSearchBar), 'Coba')
            expect(setSearchText).toBeCalled()
        })
    })
})