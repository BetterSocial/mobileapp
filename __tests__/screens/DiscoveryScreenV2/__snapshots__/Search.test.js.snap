// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Discovery Screen V2 Testing Discovery Search Bar Match Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderBottomColor": "#F5F6F7",
      "borderBottomWidth": 0.8,
      "flexDirection": "row",
      "height": 48,
      "marginTop": 0,
      "paddingVertical": 5.6000000000000005,
      "zIndex": 10,
    }
  }
>
  <View
    style={
      Object {
        "paddingLeft": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginRight": 7.2,
        }
      }
      testID="discoveryScreenBackArrow"
    >
      <View />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#F5F6F7",
          "borderRadius": 6.4,
          "flex": 1,
          "flexDirection": "row",
          "height": 28.8,
          "paddingRight": 6.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "justifyContent": "center",
            "marginLeft": 7.736000000000001,
            "marginRight": 1.336,
          }
        }
      >
        <View />
      </View>
      <TextInput
        allowFontScaling={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Search BetterSocial"
        placeholderTextColor="#C4C4C4"
        rejectResponderTermination={true}
        returnKeyType="search"
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "fontFamily": "Inter-Regular",
            "fontSize": 12.600000000000001,
            "height": 26.400000000000002,
            "marginRight": 12.8,
            "paddingBottom": 0,
            "paddingStart": 6.4,
            "paddingTop": 0,
          }
        }
        testID="discoveryScreenSearchBar"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginRight": -16.400000000000002,
            "opacity": 1,
            "paddingHorizontal": 24,
            "zIndex": 1000,
          }
        }
        testID="discoveryScreenClearButton"
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "justifyContent": "center",
            }
          }
        >
          <View />
        </View>
      </View>
    </View>
  </View>
</View>
`;
